src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -2147,7 +2147,9 @@ private function getClasses(Definition $definition): array
         $classes = [];
 
         while ($definition instanceof Definition) {
-            $classes[] = trim($definition->getClass(), '\\');
+            if ($class = $definition->getClass()) {
+                $classes[] = trim($class, '\\');
```

Comment:
maybe we can now remove `array_filter($classes)` at line 2169?

Reply:
The `array_filter` would still remove empty strings. I don't know if that's a thing here.

Reply:
removed in 40992f001b as I don't think the empty can get in (or if it does, that's because of a broken factory definition anyway)